lib/media/track/remotevideotrack.js

```diff
@@ -106,8 +110,13 @@ class RemoteVideoTrack extends RemoteMediaVideoTrack {
   attach(el) {
     const result = super.attach(el);
     this._invisibleElements.add(result);
-    this._intersectionObserver.observe(result);
-    this._resizeObserver.observe(result);
+    if (this._idleTrackSwitchOff) {
+      this._intersectionObserver.observe(result);
+    }
+
+    if (this._renderHints) {
+      this._resizeObserver.observe(result);
+    }
```

Comment:
I don't think we need the `if()` check since you are using the `NullObserver`.

Reply:
You are right that this check is not necessary, but I it helps with unit tests, as they can verify that observe being called only when option is set.

Reply:
@makarandp0 I think the unit tests can test if the correct callback (actual or fake) is called. That way, you can remove these checks.

Reply:
the unit test and the implementation both use same prototype (NullObserver) for fake, making it difficult to distinguish. maybe we can quick chat about whats the best way to do this.

Reply:
@makarandp0 Since you depend on the global `IntersectionObserver` and `ResizeObserver`, you can use two fake observers, one to verify the case where they are supported and one to verify the case where they are not supported.

```js
// Supported case
before(() => {
  global.IntersectionObserver = FakeObserver;
});

it('...', () => {
  assert(/* FakeObserver constructor is called when the RemoteVideoTrack is created */);
});

// Unsupported case
before(() => {
  delete global.IntersectionObserver;
});

it('...', () => {
  assert(/* NullObserver constructor is called when the RemoteVideoTrack is created */);
});
```

This way, you can assert that the callback is called in both cases.